Esri/calcite-web
# Documentation work for 2.0

Sorry this pr got so big. Trying to clean up everything and get the doc design in order so we're ready to dev component updates:

- updated `2.0` with latest from `master`
- cleaned up a whole bunch of doc files that were for non-existent components
- removed the `examples` folder (covered by guides now)
- updated all links to now removed example files
- tweaks to color ramps (using soon-to-be updated red values)
- tweaks to icon helper classes (small changes to spacing)
- add the description meta tag to main page layout
- ensure doc pages have a description field that is meaningful
- update home page design based on @julio8a and @bstifle 's comments (gray bg banner, tweaks to icons sizes, better gemmy, different meridian icons for two guides, removal of dark mode toggle)
- update global nav to latest version
- added the ability to render modifier class examples on a dark background (useful for `btn-white` and `text-white` which are invisible on white)
- trimmed down several sections of doc to be less verbose
- ran svgs in docs assets folder through svgo to eek out some performance
